pkg/build/builder/cmd/builder.go

```diff
@@ -226,30 +226,41 @@ func (c *builderConfig) setupGitEnvironment() (string, []string, error) {
 	return c.sourceSecretDir, bld.MergeEnv(os.Environ(), gitEnv), nil
 }
 
-// setupProxyConfig sets up a global git proxy configuration with the provided git client.
-// This is a work-around for Bug 1875639.
+// setupProxyConfig sets up a git proxy configuration with the provided git client, and directory
+// containing a local `.gitconfig` file if this directory contains a source secret.
 //
-// See also: https://bugzilla.redhat.com/show_bug.cgi?id=1875639
-func (c *builderConfig) setupProxyConfig(gitClient git.Repository) error {
+// This is a work-around for Bug 1875639 - see https://bugzilla.redhat.com/show_bug.cgi?id=1875639
+func (c *builderConfig) setupProxyConfig(gitClient git.Repository, gitConfigDir string) error {
 	gitSource := c.build.Spec.Source.Git
 	if gitSource == nil {
 		return nil
 	}
+	useLocal := false
+	if c.build.Spec.Source.SourceSecret != nil {
+		useLocal = true
+	}
 	if gitSource.HTTPProxy != nil && len(*gitSource.HTTPProxy) > 0 {
-		err := gitClient.AddGlobalConfig("http.proxy", *gitSource.HTTPProxy)
+		err := c.addGitConfig(gitClient, gitConfigDir, "http.proxy", *gitSource.HTTPProxy, useLocal)
 		if err != nil {
 			return err
 		}
 	}
 	if gitSource.HTTPSProxy != nil && len(*gitSource.HTTPSProxy) > 0 {
-		err := gitClient.AddGlobalConfig("https.proxy", *gitSource.HTTPSProxy)
+		err := c.addGitConfig(gitClient, gitConfigDir, "https.proxy", *gitSource.HTTPSProxy, useLocal)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func (c *builderConfig) addGitConfig(gitClient git.Repository, gitConfigDir string, parameter string, value string, local bool) error {
```

Comment:
s/local/useLocal/
I feel like local is to ambiguous, if it's a bool it should be something like "isLocal" or "useLocal" depending on it's usage and meaning.